### PR TITLE
Better transform error when file doesn't exist

### LINKED ISSUE #804: Unhelpful stack-trace when bam-loading a nonexistent path
NPE, no good error message:

```sh
15/08/24 02:32:14 INFO ADAMContext: Loading sorted.sam as SAM/BAM and converting to AlignmentRecords. Projection is ignored.
Exception in thread "main" java.lang.NullPointerException
	at scala.collection.mutable.ArrayOps$ofRef$.length$extension(ArrayOps.scala:114)
	at scala.collection.mutable.ArrayOps$ofRef.length(ArrayOps.scala:114)
	at scala.collection.SeqLike$class.size(SeqLike.scala:106)
	at scala.collection.mutable.ArrayOps$ofRef.size(ArrayOps.scala:108)
	at scala.collection.mutable.Builder$class.sizeHint(Builder.scala:69)
	at scala.collection.mutable.ArrayBuilder.sizeHint(ArrayBuilder.scala:24)
	at scala.collection.TraversableLike$class.builder$1(TraversableLike.scala:240)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:243)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:108)
	at org.bdgenomics.adam.rdd.ADAMContext.loadBam(ADAMContext.scala:239)
	at org.bdgenomics.adam.rdd.ADAMContext$$anonfun$loadAlignments$1.apply(ADAMContext.scala:567)
	at org.bdgenomics.adam.rdd.ADAMContext$$anonfun$loadAlignments$1.apply(ADAMContext.scala:564)
	at org.apache.spark.rdd.Timer.time(Timer.scala:57)
	at org.bdgenomics.adam.rdd.ADAMContext.loadAlignments(ADAMContext.scala:562)
	at org.bdgenomics.adam.cli.Transform.run(Transform.scala:182)
	at org.bdgenomics.utils.cli.BDGSparkCommand$class.run(BDGCommand.scala:53)
	at org.bdgenomics.adam.cli.Transform.run(Transform.scala:98)
	at org.bdgenomics.adam.cli.ADAMMain$.main(ADAMMain.scala:106)
	at org.bdgenomics.adam.cli.ADAMMain.main(ADAMMain.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:665)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:170)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:193)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:112)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
15/08/24 02:32:14 INFO SparkContext: Invoking stop() from shutdown hook
```

We can do better!

### PR DESCRIPTION
**Pull request by ryan-williams**
fixes #804

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala [hunk 1]
```diff
@@ -234,30 +235,48 @@ class ADAMContext(val sc: SparkContext) extends Serializable with Logging {
...
+            // data into each individual Read (see the argument to samRecordConverter.convert,
+            // below).
+            val samHeader = SAMHeaderReader.readSAMHeaderFrom(fp, sc.hadoopConfiguration)
+            log.info("Loaded header from " + fp)
+            val sd = adamBamDictionaryLoad(samHeader)
+            val rg = adamBamLoadReadGroups(samHeader)
+            Some((sd, rg))
+          } catch {
+            case e: Throwable => {
+              log.error(
```
Comment (ryan-williams) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala]: this bit is also in #803, should all come out in the merge-wash

**Comment by heuermh** [comment]
LGTM, should I merge using the Github button or might that cause trouble?

**Comment by ryan-williams** [comment]
Last word from @fnothaft I heard was that we can still web-merge for now; his alternate scripts etc. are not fully ready.

**Comment by heuermh** [comment]
After merging #802 and #803 via the Github merge button this one has a merge conflict.  I'll let someone with stronger git-fu chime in on how best to proceed.

**Comment by ryan-williams** [comment]
I'll resolve it and push to this branch again, 1m

**Comment by ryan-williams** [comment]
rebased; let's see what jenkins has to say

**Comment by heuermh** [comment]
Thanks @ryan-williams

**Pull request merged**